contributor-guide/modules/ROOT/pages/design-guide/dependencies.adoc

```diff
@@ -3,74 +3,46 @@
 :idprefix:
 :idseparator: -
 
-*Legacy content, currently being updated.*
+A Boost library _should generally not_ depend on other libraries, except for Boost or the pass:[C++] Standard Library. There are some exceptions. For example, boost:iostreams[] has an optional dependency on https://www.zlib.net/[*zlib*], to provide support for gzip and zlib compression and decompression. If zlib is not available, boost:iostreams[] will still work, but without the ability to handle gzip and zlib formats. Another example is boost:asio[], which has an optional dependency on https://www.openssl.org/[*OpenSSL*], to provide support for SSL and TLS protocols. These protocols are widely used for secure network communications. Again, if OpenSSL is not available, boost:asio[] will still work, but without support for SSL or TLS.
```

Comment:
>  _should generally not_

This is a little controversial because there's no underlying rule saying they can't or shouldn't depend on other libraries. For instance, Asio, which is one of the big ones, depends on OpenSSL for TSL. Boost.Compute depends on OpenCL. Beast depends on zlib for compressed HTTP communication. 

The underlying rule is that they should not have unreasonable dependencies because that would make things worse for the user and testing much more difficult. For instance, a library shouldn't depend on nlohmann/json when it can depend on boostorg/json, which boost users will already have. Any GUI library proposal would have dependencies outside boost. A library proposal for a new container type is unlikely to have extra dependencies. Dependencies also tend to be optional to avoid breaking the super-project build, although I'm not sure I completely agree with this rule.

In practice, this means most libraries end up with no dependencies outside other boost libraries. But this is a description of how things are, not how things should be unconditionally and we don't want to give this impression to prospective contributors. I'd rather describe that "A Boost library _does not generally_ depend on other libraries"...

Reply:
OK, reworded the paragraph